core/packages/contracts/contracts/SovereignTreasury.sol

```diff
@@ -0,0 +1,49 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "./Vault.sol";
+
+// TODO: transfer ownership from deployer to inbound channel
+contract SovereignTreasury is Ownable {
+    Vault public vault;
+
+    constructor(Vault _vault) {
+        vault = _vault;
+    }
+
+    // Handle a message from the bridge.
+    function handle(bytes32 _sovereignID, bytes calldata _message) external onlyOwner {
```

Comment:
Non Blocking: I do not think we want to use the `_` underscore convention anymore.
```suggestion
    function handle(bytes32 sovereignID, bytes calldata message) external onlyOwner {
```

Reply:
Agreed!

Reply:
Also `sovereignID` should be `origin` in this context. Its the parachain from whench the message originated

Reply:
ah ok. We talked about them being the same [in the doc](https://www.notion.so/snowfork/SovereignTreasury-ed3e8bf757be426b800c8918e77aa4ea?d=3c1d360cd770409c941e8d03ecc422e6#d05a7d66f68c41ec98a4f915e63082ce), so I didn't see the distinction. Is it that it's called `origin` for messages and `sovereignID` for transactions?